src/SigasiProjectCreator/ProjectCreator.py

```diff
@@ -0,0 +1,467 @@
+#     cygwin_process = subprocess.Popen(['/usr/bin/cygpath', '--windows', cygwin_path], stdout=subprocess.PIPE)
+#     cygwin_location = cygwin_process.communicate()[0].rstrip()
+#     return posixpath(cygwin_location)
+
+class ProjectCreator:
+    def __init__(self, options):
+        self.options = options
+        self.virtual_folders = [pathlib.Path('Common Libraries')]
+        self.linked_paths_simulator_layout = []
+        self.project_root = pathlib.Path(options.destination_folder).absolute()
+
+        self.sigasi_project = SigasiProject(options)
+        self.sigasi_project.unmap("/")
+
+    def check_and_create_virtual_folder(self, file_name: pathlib.Path):
+        if not isinstance(file_name, pathlib.Path):
+            raise TypeError
+        filepath = file_name.parent
+        if filepath not in self.virtual_folders:
+            new_folders = []
+            while filepath and (str(filepath) != '.') and (filepath not in self.virtual_folders):
+                new_folders.insert(0, filepath)
+                filepath = filepath.parent
+            for new_path in new_folders:
+                self.virtual_folders.append(new_path)
+                self.sigasi_project.add_link(new_path, None, True)
+
+    def check_and_create_linked_folder(self, folder_name: pathlib.Path, folder_path: pathlib.Path):
+        if not (isinstance(folder_name, pathlib.Path) and isinstance(folder_path, pathlib.Path)):
+            raise TypeError
+        # TODO future work: map some folders into Common Libraries
+        # if folder_name.startswith('dependencies'):
+        #     virtual_path_name = posixpath(os.path.join('Common Libraries', folder_name))
+        self.check_and_create_virtual_folder(folder_name)
+        self.sigasi_project.add_link(folder_name,
+                                     get_rel_or_abs_path(folder_path, self.project_root, self.options), True)
+
+    def parse_input_file(self):
+        parser = get_parser_for_type(self.options.input_format)
+        if parser is not None:
+            entries = parser(self.options.input_file, self.options)
+        else:
+            # If the parser is None, input_file contains a (list of) HDL files
+            entries = {pathlib.Path(entry).absolute().resolve(): self.options.worklib
+                       for entry in self.options.input_file}
+        return entries
+
+    def create_project(self):
+        parser_output = self.parse_input_file()
+
+        verilog_includes = None
+        verilog_defines = None
+
+        if not isinstance(parser_output, dict):
+            verilog_includes = parser_output.includes
+            verilog_defines = parser_output.defines
+            entries = parser_output.library_mapping
+            if verilog_includes is not None and len(verilog_includes) > 0:
+                verilog_includes = [pathlib.Path(include_path) for include_path in verilog_includes]
+                print("Includes: " + str(verilog_includes))
```

Comment:
What is this print? Does the user understand this? I think this should probably be behind a `-v/--verbose` flag. And it should contain more context, like 'Includes used in every file' (or what this is).

Reply:
Verbose flag + guards + better message implemented